include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.cpp

```diff
@@ -370,9 +352,19 @@ MultigridPreconditionerBase<dim, Number>::get_number_of_h_levels() const
 {
   if(data.involves_h_transfer())
   {
-    return (multigrid_variant == MultigridVariant::GlobalCoarsening ?
-              grid->coarse_triangulations.size() :
-              grid->triangulation->n_global_levels());
+    if(multigrid_variant == MultigridVariant::LocalSmoothing)
+    {
+      return grid->triangulation->n_global_levels();
+    }
+    else if(multigrid_variant == MultigridVariant::GlobalCoarsening)
+    {
+      return grid->coarse_triangulations.size() + 1;
+    }
+    else
+    {
+      AssertThrow(false, dealii::ExcMessage("Not implemented."));
+      return 0;
+    }
```

Comment:
Preparation for new multigrid coarsening sequences like "NonMatching/NonNested".